src/OpenTelemetry.Exporter.Prometheus/Implementation/PrometheusSerializerExt.cs

```diff
@@ -66,19 +66,25 @@ public static int WriteMetric(byte[] buffer, int cursor, Metric metric)
 
                     // Counter and Gauge
                     cursor = WriteMetricName(buffer, cursor, metric.Name, metric.Unit);
-                    buffer[cursor++] = unchecked((byte)'{');
 
-                    for (var i = 0; i < keys.Length; i++)
+                    int numberOfKeys = keys?.Length ?? 0;
+                    if (numberOfKeys > 0)
                     {
-                        if (i > 0)
+                        buffer[cursor++] = unchecked((byte)'{');
+
+                        for (var i = 0; i < keys.Length; i++)
                         {
-                            buffer[cursor++] = unchecked((byte)',');
+                            if (i > 0)
+                            {
+                                buffer[cursor++] = unchecked((byte)',');
+                            }
```

Comment:
@CodeBlanch Would there be benefit to moving this to above the for loop, starting `i = 1` and also adding `cursor = WriteLabel(buffer, cursor, keys[i], values[i]);` or would the compiler already do this loop unrolling?

Reply:
Oops I misread the code. another idea would be to always add the comma character after the `WriteLabel` call and then decrement the cursor after the loop so the next statement that modifies the buffer will overwrite it.

Reply:
Cool idea! Updated on https://github.com/open-telemetry/opentelemetry-dotnet/pull/2749